catalogue/webapp/components/ConfirmItemRequest/ConfirmItemRequest.tsx

```diff
@@ -184,37 +211,97 @@ const ErrorDialog: FunctionComponent<ErrorDialogProps> = ({ setIsActive }) => (
   </>
 );
 
-const ConfirmItemRequest: FunctionComponent<Props> = props => {
-  const openButtonRef = useRef<HTMLButtonElement>(null);
-  const { item, work, setIsActive, ...modalProps } = props;
-  const [isConfirmed, setIsConfirmed] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const [isError, setIsError] = useState(false); // TODO: implement this if something goes wrong with API call
+type RequestingState = undefined | 'requesting' | 'confirmed' | 'error';
```

Comment:
Would we normally have:
```suggestion
type RequestingState = 'requesting' | 'confirmed' | 'error';
```

And then reference it as `prop: RequestingState | undefined`?